src/main/java/core/basesyntax/service/impl/DeepCopyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.DeepCopy;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeepCopyImpl implements DeepCopy {
+    @Override
+    public List<FruitTransaction> getDeepCopy(List<FruitTransaction> list) {
+        List<FruitTransaction> copyList = new ArrayList<>(list.size());
+        for (FruitTransaction fruit : list) {
+            copyList.add(new FruitTransaction(fruit.getType(),
+                    fruit.getFruit(), fruit.getQuantity()));
```

Comment:
The constructor call for `FruitTransaction` seems to assume that the `FruitTransaction` class has a constructor that takes the type, fruit, and quantity. Ensure that such a constructor exists and that it indeed creates a deep copy of the object. If any of the fields of `FruitTransaction` are mutable objects themselves, you would need to create a deep copy of those fields as well.